### DISCUSSION TITLE
Modbus Device Error "transformed value out of its original type"

### DISCUSSION THREAD

**Discussion by MatteoMariano** [edgexfoundry/community#127] [Q&A]
Hi all, 
I'm trying to read values from a Modbus device based on a Modbus profile where a resource is created like this:
`{"name":"R106","description":"","attributes":{"primaryTable":"INPUT_REGISTERS","startingAddress":106},"properties":{"valueType":"Int16","readWrite":"R","defaultValue":"0","scale":"0.1"}}`

When i read from that register it returns this error:
_level=ERROR ts=2023-06-14T13:16:11.52765054Z app=device-modbus source=transform.go:66 msg="failed to transform CommandValue (DeviceResource: R106, Int16: -888): transformed value out of its original type (int16) range"_

How can I resolve this issue?

**Comment by cloudxxx8** [comment]
Use rawType and set Float32 to value type
https://docs.edgexfoundry.org/3.0/examples/Ch-ExamplesModbusdatatypeconversion/
